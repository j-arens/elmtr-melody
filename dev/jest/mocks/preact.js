module.exports = {
    unmountComponent: jest.fn(),
};
