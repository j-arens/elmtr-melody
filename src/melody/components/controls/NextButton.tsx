import BaseButton from '@components/BaseButton/';
import Icon from '@components/Icon/';
import { nextTrack } from '@redux/actions';
import { Action, State } from '@redux/type';
import { NO_OP } from '@utils/index';
import { h } from 'preact';
import { connect } from 'preact-redux';
const s = require('./styles.scss');

const mapStateToProps = (state: State) => ({
    totalTracks: state.tracks.length,
});

const mapDispatchToProps = dispatch => ({
    nextTrack: () => dispatch(nextTrack()),
});

interface Props {
    totalTracks: number;
    nextTrack: () => Action;
}

const NextButton = ({ totalTracks, nextTrack }: Props) => {
    const defaultClass = `${s.playbackCtrl} melody-playbackCtrl`;
    const disabledClass = `${defaultClass} ${s['playbackCtrl--disabled']}`;
    return (
        <BaseButton
            key="next"
            className={totalTracks > 1 ? defaultClass : disabledClass}
            onClick={totalTracks > 1 ? nextTrack : NO_OP}
        >
            <Icon className={s.playbackCtrl__icon} name="next" />
        </BaseButton>
    );
};

export default connect(mapStateToProps, mapDispatchToProps)(NextButton);
