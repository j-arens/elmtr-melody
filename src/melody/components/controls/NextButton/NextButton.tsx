import BaseButton from '@components/BaseButton/';
import Icon from '@components/Icon/';
import { WithOptionalClassName } from '@melody/components/type';
import { Action } from '@redux/type';
import { NO_OP } from '@utils/index';
import { h } from 'preact';
const s = require('../styles.scss');

export interface StateProps {
    totalTracks: number;
}

export interface DispatchProps {
    nextTrack: () => Action;
}

type Props = StateProps & DispatchProps & WithOptionalClassName;

export default ({ totalTracks, nextTrack, className = '' }: Props) => {
    const defaultClass = `${className} ${s.playbackCtrl} melody-playbackCtrl`;
    const disabledClass = `${defaultClass} ${s['playbackCtrl--disabled']}`;
    return (
        <BaseButton
            key="next"
            className={totalTracks > 1 ? defaultClass : disabledClass}
            onClick={totalTracks > 1 ? nextTrack : NO_OP}
        >
            <Icon className={s.playbackCtrl__icon} name="next" />
        </BaseButton>
    );
};
