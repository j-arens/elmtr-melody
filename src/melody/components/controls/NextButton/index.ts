import { WithOptionalClassName } from '@components/type';
import { nextTrack } from '@redux/modules/audio/actions';
import { State } from '@redux/type';
import { connect } from 'preact-redux';
import NextButton, { DispatchProps, StateProps } from './NextButton';

const mapStateToProps = (state: State) => ({
    totalTracks: state.audio.tracks.length,
});

const mapDispatchToProps = dispatch => ({
    nextTrack: () => dispatch(nextTrack()),
});

export default connect<
    StateProps,
    DispatchProps,
    WithOptionalClassName
>(mapStateToProps, mapDispatchToProps)(NextButton);
