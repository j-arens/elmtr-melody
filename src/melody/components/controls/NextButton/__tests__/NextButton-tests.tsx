import { h } from 'preact';
import { render } from 'preact-render-to-string';
import NextButton from '../NextButton';

describe('NextButton', () => {
    it('renders', () => {
        const component = render(
            <NextButton
                totalTracks={100}
                className="jet-black"
            />,
        );
        expect(component).toMatchSnapshot();
    });

    it('renders with a disabled class if totalTracks <= 1', () => {
        const component = render(
            <NextButton
                totalTracks={1}
                className="jet-black"
            />,
        );
        expect(component).toMatchSnapshot();
    });
});
