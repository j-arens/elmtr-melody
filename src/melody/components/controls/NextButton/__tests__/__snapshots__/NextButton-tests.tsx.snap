// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NextButton renders 1`] = `
<button data-cy="ctrl" class="btn jet-black playbackCtrl melody-playbackCtrl"
role="button">
  <svg class="playbackCtrl__icon" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
    <path d="M4 18l8.5-6L4 6v12zm9-12v12l8.5-6L13 6z"></path>
  </svg>
</button>
`;

exports[`NextButton renders with a disabled class if totalTracks <= 1 1`] = `
<button data-cy="ctrl" class="btn jet-black playbackCtrl melody-playbackCtrl playbackCtrl--disabled"
role="button">
  <svg class="playbackCtrl__icon" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
    <path d="M4 18l8.5-6L4 6v12zm9-12v12l8.5-6L13 6z"></path>
  </svg>
</button>
`;
